bundle/src/main/java/com/adobe/acs/commons/workflow/synthetic/impl/cq/SyntheticWorkItem.java

```diff
@@ -43,22 +49,51 @@
 
     private MetaDataMap metaDataMap = new SyntheticMetaDataMap();
 
-    public SyntheticWorkItem(final WorkflowData workflowData) {
+    private SyntheticWorkItem(final WorkflowData workflowData) {
         this.workflowData = workflowData;
         this.timeStarted = new Date();
     }
 
+    public static SyntheticWorkItem createSyntheticWorkItem(WorkflowData workflowData) {
+        return new SyntheticWorkItem(workflowData);
+    }
+
     @Override
+    public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+        String methodName = method.getName();
+        switch (methodName) {
+            case "getTimeStarted":
+                return getTimeStarted();
+            case "getTimeEnded":
+                return getTimeEnded();
+            case "getWorkflow":
+                return getWorkflow();
+            case "getNode":
+                return getNode();
+            case "getId":
+                return getId();
+            case "getWorkflowData":
+                return getWorkflowData();
+            case "getCurrentAssignee":
+                return getCurrentAssignee();
+            case "getMetaData":
+                return getMetaData();
+            case "getMetaDataMap":
+                return getMetaDataMap();
+            default:
+                LOG.error("CQ SYNTHETICWORKFLOW ITEM >> NO IMPLEMENTATION FOR {}", methodName);
+                throw new UnsupportedOperationException();
```

Comment:
Either log or throw. In this case I would suggest to throw with a meaningful message.